src/notebook/serialize.ts

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+'use strict';
+
+import {
+  ICellModel, IRawCellModel, IMarkdownCellModel, ICodeCellModel,
+  isRawCellModel, isCodeCellModel, isMarkdownCellModel
+} from '../cells';
+
+import {
+  INotebookContent, ICell, MAJOR_VERSION, MINOR_VERSION,
+  IRawCell, ICodeCell, IMarkdownCell, isRawCell, isMarkdownCell,
+  isCodeCell
+} from './nbformat';
+
+import {
+  INotebookModel
+} from './model';
+
+
+/**
+ * Serialize a notebook model.
+ */
+export 
+function serialize(nb: INotebookModel): INotebookContent {
+  let cells: ICell[] = [];
+  for (let i = 0; i < nb.cells.length; i++) {
+    let cell = nb.cells.get(i);
+    cells.push(serializeCell(cell));
+  }
+  return {
+    cells: cells,
+    metadata: nb.metadata, 
+    nbformat: MAJOR_VERSION, 
+    nbformat_minor: MINOR_VERSION 
+  };
+}
+
+/**
+ * Deserialize notebook content into a model.
+ */
+export
+function deserialize(data: INotebookContent, model: INotebookModel): void {
+  model.cells.clear();
+
+  // Iterate through the cell data, creating cell models.
+  data.cells.forEach(c => {
+    let cell: ICellModel;
+    if (isMarkdownCell(c)) {
+      cell = model.createMarkdownCell();
+    } else if (isCodeCell(c)) {
+      cell = model.createCodeCell();
+    } else if (isRawCell(c)) {
+      cell = model.createRawCell();
+    }
+    deserializeCell(c, cell);
+    this.cells.add(cell);
```

Comment:
Can we please test things before merging them, or at least try the example to see if it comes up? In this line and the next few lines, `this` should be `model`.  This completely broke the example for me (blank page, didn't even show up).